core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadWriteLDAPUserStoreManager.java

```diff
@@ -1458,6 +1458,41 @@ public void doUpdateRoleListOfUser(String userName, String[] deletedRoles, Strin
         }
     }
 
+    /**
+     * Resolves groupDN which is used in doUpdateRoleListOfUser method to modify the roles of the user.
+     *
+     * @param searchFilter Search filter.
+     * @param role         Role to be assigned to a user.
+     * @param context      LDAPRoleContext.
+     * @return String groupDN.
+     * @throws UserStoreException Error in resolving groupDN.
+     */
+    private String resolveGroupDN(String searchFilter, String role, LDAPRoleContext context) throws UserStoreException {
+
+        String roleSearchFilter = searchFilter.replace("?", escapeSpecialCharactersForFilter(role));
+        String membershipAttribute = realmConfig.getUserStoreProperty(LDAPConstants.MEMBERSHIP_ATTRIBUTE);
```

Comment:
Are we using the returning MEMBERSHIP_ATTRIBUTE attribute? If not, why do we request it?

Reply:
Yes we use that to search in the group base https://github.com/wso2/carbon-kernel/pull/2791/files#diff-25c6f9a57686933230222e3dff7b0601R1479

Reply:
The return is not used anyway ...

Reply:
Tested and fixed this with dedca0f31a6b1d54e45cc15d78aeec9a394d6116